### PR TITLE
refactor(all-inputs): refactor form-field based inputs - FE-3370

### PR DESCRIPTION
**Pull request by mkrds**
### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Full regression of all following components:

- Textbox
- Date
- Number
- GroupedCharacter
- NumeralDate
- Decimal
- Select
- MultiSelect
- FilterableSelect
- RadioButton
- Checkbox
- Switch

### CODE REVIEW

File: src/components/switch/switch.component.js [hunk 1]
```diff
@@ -154,6 +211,7 @@ Switch.defaultProps = {
   labelInline: false,
   reverse: true,
   validationOnLabel: false,
+  "data-component": "Switch",
```
Comment (sjohnson-sage) [inline src/components/switch/switch.component.js:214]:
```suggestion
  "data-component": "switch",
```

File: src/components/switch/switch.stories.mdx [hunk 2]
```diff
@@ -312,3 +315,5 @@ For more information check our [Validations](?path=/docs/documentation-validatio
 ### Switch
 
 <StyledSystemProps of={Switch} noHeader margin />
+
+Any other props supplied will be provided to the underlying html input element
```
Comment (sjohnson-sage) [inline src/components/switch/switch.stories.mdx:319]:
```suggestion
Any other props supplied will be provided to the underlying HTML input element.
```

File: src/components/textarea/textarea.stories.mdx [hunk 3]
```diff
@@ -272,3 +272,5 @@ For more information check our [Validations](?path=/docs/documentation-validatio
 ### Textarea
 
 <StyledSystemProps of={OriginalTextarea} noHeader margin />
+
+Any other props supplied will be provided to the underlying html input element
```
Comment (sjohnson-sage) [inline src/components/textarea/textarea.stories.mdx:276]:
```suggestion
Any other props supplied will be provided to the underlying HTML input element.
```

File: src/components/textbox/textbox.stories.mdx [hunk 4]
```diff
@@ -270,3 +270,5 @@ For more information check our [Validations](?path=/docs/documentation-validatio
 ### Textbox
 
 <StyledSystemProps of={OriginalTextbox} noHeader margin />
+
+Any other props supplied will be provided to the underlying html input element
```
Comment (sjohnson-sage) [inline src/components/textbox/textbox.stories.mdx:274]:
```suggestion
Any other props supplied will be provided to the underlying HTML input element.
```

**Review by sjohnson-sage — COMMENTED** [review]
LGTM in general, mostly minor comments.

I'm concerned about adding `className` props back in, we took those out purposely to stop consumers using custom CSS over our components. What was the reason for adding these back in?

**Comment by mkrds** [comment]
@sjohnson-sage Thanks for the review! I'm not entirely sure about the className to be honest. I decided to list it on all the input components because we use `className` internally to style `Select` (so `Select` component has to accept it) component in `Pager` component so I thought it's best to be consistent.
How would you approach it? Do we discard this prop at all? Or simply pass it along the other props to the underlying HTML `input` element?

**Comment by sjohnson-sage** [comment]
> @sjohnson-sage Thanks for the review! I'm not entirely sure about the className to be honest. I decided to list it on all the input components because we use `className` internally to style `Select` (so `Select` component has to accept it) component in `Pager` component so I thought it's best to be consistent.
> How would you approach it? Do we discard this prop at all? Or simply pass it along the other props to the underlying HTML `input` element?

Hi @mkrds, I don't think we should add it back in because we don't want to encourage consuming apps to write their own styled components over our components, which (I think) was why we took them out in the first place.

If we need to use the `className` prop internally for our own components I think we should just make it a private prop and not document it.

Does that sound ok to you? Happy to open this up to the team and have a discussion about it if you want to 👍🏻

File: src/__internal__/form-field/form-field.component.js [hunk 5]
```diff
@@ -60,54 +63,13 @@ const FormField = ({
   // the behaviour
```
Comment (edleeks87) [inline src/__internal__/form-field/form-field.component.js:63]: Can these comments not be removed now?

**Review by edleeks87 — COMMENTED** [review]
If we do need to re-add the `className` prop then I agree with @sjohnson-sage we should mark them as `@private` `@ignore`

File: src/components/select/filterable-select/filterable-select.stories.mdx [hunk 6]
```diff
@@ -592,11 +599,16 @@ If there is no `id` prop specified on an object, then the exact objects will be
 
 ## Props
 
-### Filterable Select 
+### Filterable Select
+
 <StyledSystemProps of={FilterableSelect} noHeader margin />
 
-### Props derived from the Textbox Component
+Due to the `Textbox` component being used internally by the `FilterableSelect` component, most of the `Textbox` props are appliable to `FilterableSelect`
```
[the remark this answers is not in the document]
Comment (sjohnson-sage) [reply src/components/select/filterable-select/filterable-select.stories.mdx:609]: This is still outstanding

File: src/components/select/multi-select/multi-select.stories.mdx [hunk 7]
```diff
@@ -480,10 +480,12 @@ See <LinkTo kind='Documentation/Colors' name="page">Colors</LinkTo> for more inf
 
 <StyledSystemProps of={MultiSelect} noHeader margin />
 
-### Props derived from the Textbox Component
+Due to the `Textbox` component being used internally by the `MultiSelect` component, most of the `Textbox` props are appliable to `MultiSelect`
```
[the remark this answers is not in the document]
Comment (sjohnson-sage) [reply src/components/select/multi-select/multi-select.stories.mdx:486]: and this

File: src/components/select/simple-select/simple-select.stories.mdx [hunk 8]
```diff
@@ -596,11 +605,16 @@ The inline label can change to be top aligned at a breakpoint. Enable this by pa
 
 ## Props
 
-### Select 
+### Select
+
 <StyledSystemProps noHeader of={Select} margin />
 
-### Props derived from the Textbox Component
+Due to the `Textbox` component being used internally by the `SimpleSelect` component, most of the `Textbox` props are appliable to `SimpleSelect`
```
[the remark this answers is not in the document]
Comment (sjohnson-sage) [reply src/components/select/simple-select/simple-select.stories.mdx:615]: and this

**Review by sjohnson-sage — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by sianford — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by sianford — APPROVED** [review]

**Pull request merged**